http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcClientConfiguration.java

```diff
@@ -237,6 +272,26 @@ protected void resolveUrls() {
         }
     }
 
+    private JsonWebKeySet createrealmKeySet() throws Exception{
+        HttpGet request = new HttpGet(jwksUrl);
```

Comment:
Can we make use of the Oidc.sendJsonHttpRequest method here to simplify the logic below?

Reply:
I tried doing this and kept getting a class mismatch for the `JsonSerialization.readValue()` function. 
I tried this for the JsonWebKeySet, String, and StringBuilder classes. 
When I was initially writing this function, I was also trying to use `JsonSerialization.readValue()` and did not succeed. Which is why I had to do this workaround.

Reply:
There should be an example in JWKPublicKeyLocator where we call sendJsonHttpRequest and store the result in a JsonWebKeySet. Did you take a look there to see how that differs from what you were trying to do?

Reply:
I see what the difference is. 
The JWKPublicKeyLocator class uses org.wildfly.security.jose.jwk.JsonWebKeySet and I am using the Jose4j one, which is why the parsing failed before. 
I have switched it to the wildfly.security one now and it works. 
Thank you!